src/indexing.c

```diff
@@ -301,3 +393,65 @@ ts_indexing_find_clustered_index(Oid table_relid)
 
 	return index_relid;
 }
+
+typedef enum IndexValidity
+{
+	IndexInvalid = 0,
+	IndexValid,
+} IndexValidity;
+
+static bool
+ts_indexing_mark_as(Oid index_id, IndexValidity validity)
```

Comment:
why not just `bool index_is_valid` instead of enum? Since this is all static I think it's cleaner.

Reply:
or do we forsee other states?

Reply:
this is more readable at the call sight.